intermediate/challenge-1/src/main/java/com/audela/challenge/busapi/controller/UnsecureController.java

```diff
@@ -0,0 +1,43 @@
+package com.audela.challenge.busapi.controller;
+
+import com.audela.challenge.busapi.entity.DriverEntity;
+import com.audela.challenge.busapi.util.JwtUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Use this controller to get tokens for accessing the api for testing.
+ */
+@RestController
+@RequestMapping("/unsecure")
+public class UnsecureController {
+
+    @Autowired
+    JwtUtils jwtUtils;
+    @GetMapping(value = "/get-employee-token")
```

Comment:
- in a real application how would you implement this mechanism  in a secure way ? 
- what would all the considerations to take in count  for a real app ?

Reply:
One way is to validate the user with username and password then crate a token and return.
Other options is to create or use a separate service for token generation. Usually organisations use a common authentication provider for all the applications.